scripts/gen_song_schema.py

```diff
@@ -0,0 +1,300 @@
+                template[prop] = 0.0
+            elif prop_type == 'boolean':
+                template[prop] = False
+            elif prop_type == 'array':
+                if 'items' in prop_schema and 'enum' in prop_schema['items']:
+                    template[prop] = [f"<{prop}>"]
+                else:
+                    template[prop] = [generate_template(prop_schema['items'])]
+            elif prop_type == 'object':
+                template[prop] = generate_template(prop_schema)
+    return template
+
+
+def update_schema(schema, top_class, file_type_enum, data_type_enum):
+    # Update the name property within the analysisType property to be a constant value from top_class
+    if 'analysisType' in schema.get('properties', {}):
+        if 'properties' in schema['properties']['analysisType']:
+            schema['properties']['analysisType']['properties']['name'] = {"const": top_class}
+    
+    # Update the fileType and dataType properties to be enum lists from file_type_enum and data_type_enum
+    if 'files' in schema.get('properties', {}):
+        if 'items' in schema['properties']['files']:
+            if 'properties' in schema['properties']['files']['items']:
+                if 'fileType' in schema['properties']['files']['items']['properties']:
+                    schema['properties']['files']['items']['properties']['fileType']['enum'] = file_type_enum
+                if 'dataType' in schema['properties']['files']['items']['properties']:
+                    schema['properties']['files']['items']['properties']['dataType']['enum'] = data_type_enum
+    return schema
+
+def ensure_directory_exists(directory):
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+def process_schema(input_file, top_class, options, schema_dir, example_dir):
+    
+    # Generate JSON schema using linkml generate command with --top-class option
+    json_schema_str = generate_json_schema(input_file, top_class)
+
+    # Resolve references in the JSON schema
+    schema_with_refs = resolve_references(json_schema_str)
+
+    # Create a copy of the resolved schema
+    schema_copy = copy.deepcopy(schema_with_refs)
+
+    # Drop specific fields from the JSON schema
+    fields_to_drop = ["$id", "$defs", "description", "version", "title",
+                      "additionalProperties", "metamodel_version"]
+    schema_copy_drop = drop_fields(schema_copy, fields_to_drop)
+
+    # Update the name property within the analysisType property
+    file_type_enum = options.get('options', []).get('fileTypes', [])
+    data_type_enum = options.get('dataType', []) 
+    schema_copy_drop = update_schema(schema_copy_drop, top_class, file_type_enum, data_type_enum)
+
+    # Remove all "title" fields from the schema
+    remove_titles(schema_copy_drop)
+
+    # Create a copy of the resolved schema
+    schema_copy_drop2 = copy.deepcopy(schema_copy_drop)
+    schema_copy_drop3 = copy.deepcopy(schema_copy_drop)
```

Comment:
Could we provide these with more informed names e.g. `schema_copy_dynamic` instead of `schema_copy_drop2`